sympy/tensor/array/ndim_array.py

```diff
@@ -377,19 +377,29 @@ def __sub__(self, other):
 
     def __mul__(self, other):
         from sympy.matrices.matrices import MatrixBase
+        from sympy.tensor.array import SparseNDimArray
 
         if isinstance(other, (Iterable, NDimArray, MatrixBase)):
             raise ValueError("scalar expected, use tensorproduct(...) for tensorial product")
+
         other = sympify(other)
+        if isinstance(self, SparseNDimArray):
```

Comment:
both `self` and `other` have to be sparse. Remember? We decided that mixing sparse and dense results in a dense array. Do you prefer to change this behaviour?